monday/graphqlclient/client.py

```diff
@@ -15,27 +17,43 @@ def inject_token(self, token, headername='Authorization'):
         self.token = token
         self.headername = headername
 
+    def inject_headers(self, headers):
+        self.headers = headers
+
     def _send(self, query, variables):
         payload = {'query': query}
-        headers = {}
+        headers = self.headers
         files = None
-
+        
         if self.token is not None:
             headers[self.headername] = '{}'.format(self.token)
 
         if variables is None:
-            headers['Content-Type'] = 'application/json'
+            if 'Content-Type' not in headers:
+                headers['Content-Type'] = 'application/json'
+            
             payload = json.dumps({'query': query}).encode('utf-8')
+        
         elif variables.get('file', None) is not None:
-            headers['content'] = 'multipart/form-data'
+            if 'content' not in headers:
+                headers['content'] = 'multipart/form-data'
+                
             files = [
                 ('variables[file]', (variables['file'], open(variables['file'], 'rb')))
             ]
 
         try:
             response = requests.request("POST", self.endpoint, headers=headers, data=payload, files=files)
             response.raise_for_status()
+            self._catch_error(response)
             return response.json()
         except requests.HTTPError as e:
             print(e)
             raise e
+        except MondayQueryError as ex:
```

Comment:
[non-blocking] This could be collapsed into the previous except block using a tuple, i.e. `except (requests.HTTPError, MondayQueryError) as e`.